pkg/apiserver/diagnose/templates.go

```diff
@@ -17,182 +17,11 @@ import (
-        }
-      </script>
-    </body>
-  </html>
-{{ end }}
-`
-
-const TemplateTable = `
-{{ define "sql-diagnosis/table" }}
-  <div class="report-container">
-    {{ range $cIdx, $c := .Category }}
-      {{ if eq $cIdx 0 }}
-        {{ if . }}
-          <h1 class="title is-size-2">{{.}}</h1>
-        {{ end }}
-      {{ else if eq $cIdx 1 }}
-        {{ if . }}
-          <h1 class="title is-size-3">{{.}}</h1>
-        {{ end }}
-      {{ else if eq $cIdx 2 }}
-        {{ if . }}
-          <h1 class="title is-size-4">{{.}}</h1>
-        {{ end }}
-      {{ else }}
-        {{ if . }}
-          <h1 class="title is-size-5">{{.}}</h1>
-        {{ end }}
-      {{ end }}
-    {{ end }}
-    <h3 class="is-size-4">{{ .Title }}</h3>
-    {{ if .CommentEN }}
-      <p>{{ .CommentEN }}</p>
-    {{ end }}
-    <table class="table is-bordered is-hoverable is-narrow is-fullwidth">
-      <thead>
-        <tr>
-          {{ range .Column }}
-            <th>{{ . }}</th>
-          {{ end }}
-        </tr>
-      </thead>
-      <tbody>
-        {{ range $rowIdx, $row := .Rows }}
-          {{ $len := len .SubValues }}
-          <tr>
-            {{ range $vIdx, $v := .Values }}
-              <td>
-                {{ . }}
-                {{ if and ($row.Comment) (eq $vIdx 0) }}
-                  <div class="dropdown is-hoverable is-up">
-                    <div class="dropdown-trigger">
-                      <span class="icon has-text-info">
-                        <i class="fas fa-info-circle"></i>
-                      </span>
-                    </div>
-                    <div class="dropdown-menu">
-                      <div class="dropdown-content">
-                        <div class="dropdown-item">
-                          <p>{{$row.Comment}}</p>
-                        </div>
-                      </div>
-                    </div>
-                  </div>
-                {{ end }}
-                {{ if and (gt $len 0) (eq $vIdx 0)}}
-                  &nbsp;&nbsp;&nbsp;
-                  <a href="javascript:void(0)"
-                     class="subvalues-toggle"
-                     data-row="{{$rowIdx}}">expand</a>
-                {{ end }}
-              </td>
-            {{ end }}
-          </tr>
-          {{ if gt $len 0 }}
-            {{ range .SubValues }}
-              <tr class="subvalues fold" data-row="{{$rowIdx}}">
-                {{ range $cIdx, $v := . }}
-                  {{ if eq $cIdx 0 }}
-                    <td>|-- {{ . }}</td>
-                  {{ else }}
-                    <td>{{ . }}</td>
-                  {{ end }}
-                {{ end }}
-              </tr>
-            {{ end }}
-          {{ end }}
-        {{ end }}
-      </tbody>
-    </table>
-  </div>
-{{ end }}
-`
+func FetchTemplates() []utils.TemplateInfoWithFilename {
+	var result []utils.TemplateInfoWithFilename
+	result = append(result, readFromFS("sql-diagnosis/index", "pkg/apiserver/diagnose/templates/index.gohtml"))
```

Comment:
Is it possible to support adding new templates?

Reply:
Of cause yes. Just duplicating this line and change template name and filename.

Reply:
Hmm, I mean if new templates are added on the disk, can we support tracking them without reboot the app? For example, as you can see, `sql-diagnosis/table` template is referenced by `sql-diagnosis/index` template, so if newly added templates are not tracked, then what we can do with existing template is limited.

Reply:
Oh, it can not right now. I will improve this!